### PR TITLE
Post feed

### PR DESCRIPTION
**Pull request by TylerSchaus**
**Post Feed PR**
PR includes most key functionalities of the **Post Feed** .

**Updated** 

FeedFragment:
- Loading feed from the `PostListManager`
- Viewing the feed. 
- Liking / disliking a post, updates appropriate fields in `Post`. 
- Viewing the comment section. 
- Making a new comment, updates `Comments` in `Post`. 

FeedViewModel:
- Updated strictly for testing purposes (manually adding 'fake' posts and adding examples fields to test the display). 

**Added:** 

- comment_item.xml 
- post_item.xml
- discussioin_post_item.xml 
- filter_view.xml 

**Still to complete:** 

- Functionality for deleting a comment. Currently the "Delete" button appears on all comments. Will be updated to only appear when comment `userId` = current user `userId`. Also still waiting to add functionality so that deleting a comment updates `comments` in `Post` 
- Reporting a post. 

**For reviewers:** 
Nothing in particular, just ensure this should be compatible with your parts. 
Ian: Mostly just included you because you designed the data classes and are working on a "User list" of some kind. Hopefully you can make that is compatible with the current set up.

### CODE REVIEW

**Comment by TylerSchaus** [comment]
Forgot to mention both image fields (poster profile picture / commenter profile picture) are yet to be added as well.

File: app/src/main/java/com/example/cosc341_project/MainActivity.java [hunk 1]
```diff
@@ -1,6 +1,11 @@
 package com.example.cosc341_project;
 
 import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
```
[the remark this answers is not in the document]
Comment (TylerSchaus) [reply app/src/main/java/com/example/cosc341_project/MainActivity.java:8]: Not sure what you're talking about here, all are required for the program.

[the remark this answers is not in the document]
Comment (TylerSchaus) [reply app/src/main/java/com/example/cosc341_project/MainActivity.java:8]: and line 17 is blank and line 20 import is also necessary

**Review by iansteyn — CHANGES_REQUESTED** [review]
# Ian's Review
_...which disappeared for no reason.... and which I have rewritten below._

## ✅ What works
Overall, great work. UI looks good, clean, is usable. It loads quickly and doesn't lag/catch; the scrolling is very smooth. Likes/dislikes work (properly; user can't add both like and dislike which is good) adding comments works, filtering works. Well done.

## ❗Requested Changes
You have to save changes (comments, likes/dislikes) to the file. I suggest you do this by calling `plm.saveToFile()` inside of `onDestroyView` at the end of `FeedFragment`.

[more incoming, I just want to save my progress so it doesn't die on me again]

File: app/src/main/java/com/example/cosc341_project/MainActivity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (iansteyn) [reply app/src/main/java/com/example/cosc341_project/MainActivity.java:8]: lol ok I must be buggin out then

File: app/src/main/java/com/example/cosc341_project/ui/feed/FeedViewModel.java [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
+        String [] sightingPostTags = {"sasquatch", "sighting"};
+
+        Post tempPost = plm.postList.get(0);
+        tempPost.addComment(123, "This is an example comment");
+        plm.postList.add(new SightingPost(123, "Sighting post", "Example description", sightingPostTags, "nullimage", "Kelowna, BC"));
+
+    }
+
+    public PostListManager getPostListManager() {
+        return plm;
```
[the remark this answers is not in the document]
Comment (TylerSchaus) [reply app/src/main/java/com/example/cosc341_project/ui/feed/FeedViewModel.java:32]: Very true I just had it set this way for testing before you finished data classes with that method passing a test list of posts but now that data classes are set can be removed.

[the remark this answers is not in the document]
Comment (iansteyn) [reply app/src/main/java/com/example/cosc341_project/ui/feed/FeedViewModel.java:32]: Ok, can you change this before merge?

**Review by iansteyn — COMMENTED** [review]
Requested changes fixed. I will resolve merge conflicts then approve the merge.

**Pull request merged**